R/plotreg.R

```diff
@@ -1,394 +1,372 @@
+                       scales = "free_y")
+        
+        if (length(custom.title) > 0) {
+            p <- p +   ggtitle(custom.title)
+        } else if (length(models) == 1) {
+            p <- p +   ggtitle("Model")
+        } else if (length(models) > 1) {
+            p <- p +  ggtitle("Models")
+        }
+        
+    } else if (type == "forest") {
+        
+        if (length(reorder.coef) > 0) {
+            #reorder levels 
+            dataframe$co.names <- ordered(dataframe$co.names, levels = (unique(as.character(dataframe$co.names))))
+            #reorder coef
+            dataframe$co.names <- factor(dataframe$co.names, levels(dataframe$co.names)[reorder.coef])
+        } else {
+            #reorder levels 
+            dataframe$co.names <- ordered(dataframe$co.names, levels = (unique(as.character(dataframe$co.names))))
+            #reorder coef in original order
+            startlevel<- c(1:length(dataframe$co.names))
+            dataframe$co.names <- factor(dataframe$co.names, levels(dataframe$co.names)[startlevel])
+        }
+        
+        p <- ggplot(dataframe, aes(lab, co)) + 
+            geom_hline(yintercept=0, 
+                       lty=2, 
+                       lwd=1, 
+                       colour="grey50") +
+            geom_errorbar(aes(ymin=lower.outer, ymax=upper.outer), 
+                          lwd=1, 
+                          colour = ifelse(dataframe[order(dataframe$co.names, dataframe$co), ]$signif == TRUE , signif.light, insignif.light), 
+                          width=0) +
+            geom_errorbar(aes(ymin= lower.inner, ymax= upper.inner), 
+                          lwd=2.5, 
+                          colour = ifelse(dataframe[order(dataframe$co.names, dataframe$co), ]$signif == TRUE, signif.medium, insignif.medium),
+                          width=0) +
+            geom_point(size=3, 
+                       pch = ifelse(dataframe[order(dataframe$co.names, dataframe$co), ]$signif == TRUE, 21, 22), 
+                       fill = ifelse(dataframe[order(dataframe$co.names, dataframe$co), ]$signif == TRUE, signif.dark, insignif.dark)) +
+            coord_flip() + 
+            theme_bw() +
+            xlab(" ") +
+            facet_wrap(~co.names, 
+                       strip.position="left", 
+                       nrow=length(dataframe), 
+                       scales = "free_y")
+        
+        if (length(custom.title) > 0) {
+            p <- p +   ggtitle(custom.title)
+        } else if (length(models) == 1) {
+            p <- p +   ggtitle("Model")
+        } else if (length(models) > 1) {
+            p <- p +  
+                ggtitle("Models")
+        }
+    } 
+    
+    # adds message to p as ylab; adds output meassages to paste that comes as R output (not in plot)
+    if (use.se[i] == TRUE && length(models[[i]]@se) > 0) {
+        p <- p + ylab("Bars denote SEs.\n Circle points and red color \n denote significance.")
```

Comment:
Ideally, it would be possible to let the user override the red and blue colours by his or her preferred colours. For example, for publication purposes, it may make sense to just use black.